react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -1,20 +1,64 @@
-import React from 'react';
+import React, { Component } from 'react';
+
+import { calculateWinner } from '../utils';
 
 import styles from './styles.module.scss';
 import Board from './components/Board';
 
-function Game () {
-  return (
-    <div className={styles.game}>
-      <div className={styles.gameBoard}>
-        <Board />
-      </div>
-      <div className={styles.gameInfo}>
-        <div>{/* status */}</div>
-        <ol>{/* TODO */}</ol>
+class Game extends Component {
+  state = {
+    history: [
+      {
+        squares: Array(9).fill(null)
+      }
+    ],
+    xIsNext: true
+  };
+
+  handleClick = i => {
+    const { history } = this.state;
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+
+    if (calculateWinner(squares) || squares[i]) {
+      return;
+    }
+
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+
+    this.setState({
+      history: history.concat([
+        {
+          squares
+        }
+      ]),
+      xIsNext: !this.state.xIsNext
+    });
+  };
+
+  render() {
+    const { history } = this.state;
+    const current = history[history.length - 1];
+    const winner = current && calculateWinner(current.squares);
```

Comment:
You are calculating the winner in each render. You should be calculating this in each click, and save the component state to prevent calculating it again.
When you do a click, you could save in the winner in the state, and use it from there in the render.
In handleClick, you would change the calculateWinner condition for just checking if there is a winner in this.state.winner. That way, you would only be calculating the winner once per click.